uwblueprint/community-fridge-kw
# feat: initial donor dropoff view

## Notion ticket link

[Donor Dropoff View](https://www.notion.so/uwblueprintexecs/Donor-View-Upcoming-Dropoffs-Edit-Cancel-58b5ca107a0a49618d35bcf68e8a1e17)



## Implementation description
* Moved the `<Header/>` component to be mounted in `App.tsx` to avoid rerenders
* Added Donor dropoff view at `/dropoff` and the donor schedule details at `/dropoff/:id?donorId=`
* Created cancel dropoff modal
* Add `getDonorByUserId` endpoint: https://github.com/uwblueprint/community-fridge-kw/pull/38
* All functionality should be intact, UI still needs work


https://user-images.githubusercontent.com/19617248/141862607-29e54fe7-b34d-44c8-b687-e406749a631e.mp4
<img width="361" alt="Screen Shot 2021-11-15 at 5 26 18 PM" src="https://user-images.githubusercontent.com/19617248/141862667-1329a39e-9ea3-41c0-9223-9e34b9dc77a9.png">
<img width="362" alt="Screen Shot 2021-11-15 at 5 26 26 PM" src="https://user-images.githubusercontent.com/19617248/141862673-5a63737c-f776-41a5-b175-57d4f6c1f296.png">
<img width="357" alt="Screen Shot 2021-11-15 at 5 26 33 PM" src="https://user-images.githubusercontent.com/19617248/141862676-920cd31f-dc65-4472-a23a-ae4afc3970cf.png">



https://user-images.githubusercontent.com/19617248/141862717-556ebc52-70cc-492d-aaa5-47b86384c7c8.mp4



## Steps to test
1. Create a donor, login as donor and `POST`  a few schedules in order to see it populated, test out buttons: 
```json
    {
       "donorId": 2,
     "categories": ["Non-perishables"],
       "size": "large",
      "pickupLocation": "University of Waterloo",
      "startTime": "Mon Nov 15 2021 22:19:23",
      "endTime": "Tues Nov 16 2021 22:19:25",
      "status": "Pending",
      "isPickup": true,
      "volunteerNeeded": true,
      "frequency": "Weekly on Monday",
      "notes": "These are copied notes"
    }
```



## What should reviewers focus on?
* The UI isn't fully accurate but make sure to test out functionality
